### PR TITLE
Provide a Virtual Method Anchor for Classes in Headers 

### PR DESCRIPTION
**Pull request by mjcarroll**
# 🦟 Bug fix

Based on https://github.com/gazebosim/gz-rendering/issues/702#issuecomment-1217673602

## Summary

We have some classes defined in headers that need to have at least one virtual method implemented in a `cc` file.

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by ahcorde — CHANGES_REQUESTED** [review]
```
  /github/workspace/include/gz/rendering/ArrowVisual.hh:35:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
  /github/workspace/include/gz/rendering/BoundingBox.hh:53:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
```

**Comment by mjcarroll** [comment]
@osrf-jenkins retest this please

File: include/gz/rendering/Grid.hh [hunk 1]
```diff
@@ -35,7 +35,7 @@ namespace gz
       public virtual Geometry
     {
       /// \brief Destructor
-      public: virtual ~Grid() { }
+      public: virtual ~Grid() = 0;
```
Comment (iche033) [inline include/gz/rendering/Grid.hh:38]: should this be pure virtual?

Comment (mjcarroll) [reply include/gz/rendering/Grid.hh]: It should not, rolled back.

**Review by iche033 — APPROVED** [review]

**Pull request merged**